core/src/main/scala/org/ensime/core/AnalyzerManager.scala

```diff
@@ -73,44 +74,46 @@ class AnalyzerManager(
       withExistingModuleFor(fileInfo, req)((req, moduleId) =>
         getOrSpawnNew(moduleId) forward req)
     case req @ TypecheckFilesReq(files) =>
-      if (files.exists(config.findProject(_).isEmpty))
-        sender ! EnsimeServerError("Update .ensime file.")
-      else {
-        val original = sender
-        val filesPerProject = files.groupBy(config.findProject(_)).map(x => x._1.get -> x._2)
+      val original = sender
+      val filesPerProject = files.groupBy(config.findProject(_)).map(x => x._1 -> x._2)
+
+      context.actorOf(Props(new Actor {
+        private var remaining = filesPerProject.size
+        private var aggregate: List[String] = List.empty
 
-        context.actorOf(Props(new Actor {
-          private var remaining = filesPerProject.size
-          private var aggregate: List[String] = List.empty
+        override def preStart: Unit =
+          for ((optionalModuleId, list) <- filesPerProject)
+            optionalModuleId match {
+              case Some(moduleId) =>
+                getOrSpawnNew(moduleId) ! TypecheckFilesReq(list)
```

Comment:
I still thing `getOrSpawnNew` should return `sauron` if it doesn't match, which means it doesn't return `Option` anymore.